File: mobile/modules/barnabi-card-expand/android/src/main/java/expo/modules/barnabicardexpand/CardItemRecord.kt
package expo.modules.barnabicardexpand

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

/**
 * JS → Kotlin bridge record for a single card. Mirror of iOS's
 * `CardItemRecord.swift` and JS's `CardItem` type. Fields must stay
 * synchronized across all three; there is no shared schema.
 *
 * `Record` is Expo's equivalent of a Codable/Decodable data class. At prop
 * decode time, expo-modules-core takes the JS-side dictionary and populates
 * the `@Field`-annotated properties. Missing optional fields stay null;
 * missing non-null fields fall back to the declared default.
 */
class CardItemRecord : Record {
  @Field var id: String = ""
  @Field var title: String = ""
  @Field var subtitle: String? = null
  @Field var imageUrl: String? = null
  @Field var badge: String? = null
  @Field var accentColor: String? = null
}

File: mobile/modules/barnabi-card-expand/ios/CardItemRecord.swift
import ExpoModulesCore

/**
 * JS → Swift bridge record for a single card. Expo Modules Core maps JS
 * dictionaries onto these `Record` types at prop decode time. Fields declared
 * with `@Field` are read from the matching JS key; absent optional fields
 * decode to `nil`, absent required fields fall back to the `= ""` default.
 *
 * If you add a field here you must also add it to `CardItem` in the
 * module's `index.tsx` and to the Kotlin `CardItemRecord.kt`. All three must
 * stay in sync; there is no code-gen yet for this module.
 */
struct CardItemRecord: Record, Identifiable {
  @Field var id: String = ""
  @Field var title: String = ""
  @Field var subtitle: String?
  @Field var imageUrl: String?
  @Field var badge: String?
  @Field var accentColor: String?
}
